test/spanner.ts

```diff
@@ -3211,6 +3326,19 @@ describe('Spanner with mock server', () => {
       assert.ok(!beginTxnRequest, 'beginTransaction was called');
     });
 
+    it('should catch an exception error during invalid queries while using inline begin transaction', async () => {
+      const database = newTestDatabase();
+      await database.runTransactionAsync(async tx => {
+        try {
+          await Promise.all([tx!.run(selectSql), tx!.run(invalidSql)]);
```

Comment:
Do we already have a test for the same with these requests in the other order? That is; with the invalid SQL first?

If not, can we add that as well, and verify that:
1. The initial call to ExecuteStreamingSql fails, which also means that there is no transaction ID that is returned.
2. The transaction is retried with an explicit BeginTransaction and then the invalid SQL + working SQL statement.
3. Catch the error from the invalid SQL statement.
4. Commit the transaction (even though one of the statements failed).
5. The result should be (I think):

```
ExecuteSql(transaction{begin: {readwrite=true}})
BeginTransaction(readwrite=true)
ExecuteSql(transaction{id:...})
ExecuteSql(transaction{id:...})
Commit
```

Reply:
Will be doing this in a separate PR